### PR TITLE
Add Ed25519 to Curve25519 conversion

### PR DESCRIPTION
**Pull request by benma**
Issue #19

### CODE REVIEW

**Comment by benma** [comment]
Please let me know if there is anything I can improve.

**Comment by benma** [comment]
Bump. This is still relevant.

File: src/main/java/org/abstractj/kalium/keys/SigningKey.java [hunk 1]
```diff
@@ -60,6 +60,12 @@ public VerifyKey getVerifyKey() {
         return this.verifyKey;
     }
 
+    public PrivateKey getPrivateKey() {
+        byte []secretKey = zeros(CRYPTO_SCALARMULT_CURVE25519_BYTES);
+        sodium().crypto_sign_ed25519_sk_to_curve25519(secretKey, seed);
```
Comment (abstractj) [inline src/main/java/org/abstractj/kalium/keys/SigningKey.java]: Why you would like to expose the private key? Also, I suggest to change the method name to something like `convertToCurve`.

Just doing it by default feels wrong

File: src/test/java/org/abstractj/kalium/fixture/TestVectors.java [hunk 2]
```diff
@@ -105,4 +106,5 @@
     public static final String SIGN_SIGNATURE = "6bd710a368c1249923fc7a1610747403040f0cc30815a00f9ff548a896bbda0b" +
             "4eb2ca19ebcf917f0f34200a9edbad3901b64ab09cc5ef7b9bcc3c40c0ff7509";
     public static final String SIGN_PUBLIC = "77f48b59caeda77751ed138b0ec667ff50f8768c25d48309a8f386a2bad187fb";
+    public static final String SIGN_PUBLIC_PUBLIC = "35488a98f7ec26ae27099809afb27587b198b1197b5bcb0dec41153db2bf9952";
```
Comment (abstractj) [inline src/test/java/org/abstractj/kalium/fixture/TestVectors.java]: What does PUBLIC_PUBLIC means? Maybe we should rename it to something like `PUBLIC_CURVE25519`?

**Comment by abstractj** [comment]
@benma overall the PR looks good, but I added some suggestions.

File: src/main/java/org/abstractj/kalium/keys/VerifyKey.java [hunk 3]
```diff
@@ -41,6 +42,12 @@ public VerifyKey(String key, Encoder encoder) {
         this(encoder.decode(key));
     }
 
+    public PublicKey getPublicKey() {
+        byte []publicKey = zeros(CRYPTO_SCALARMULT_CURVE25519_BYTES);
+        sodium().crypto_sign_ed25519_pk_to_curve25519(publicKey, key);
+        return new PublicKey(publicKey);
```
Comment (abstractj) [inline src/main/java/org/abstractj/kalium/keys/VerifyKey.java]: The same here, maybe we rename the method to `convertToCurve`? I believe that in the context of libsodium 25519 is implicit and we can add this to the javadocs.

**Comment by abstractj** [comment]
@blavenie I really appreciate you taking the time to make this patch. However like we discussed, there are some changes to be done. I'm going to close this, but would love to see you open a new PR with the changes that take the direction we discussed. Thank you!

**Pull request closed without merging**